src/components/Table/Table.types.part.tsx

```diff
@@ -29,6 +29,25 @@ export interface TableCellEvent<T> {
   data?: T;
 }
 
+/**
+ * Arguments for sorting events.
+ */
+export interface TableSortEvent {
+  /**
+   * The index of the column that's sorted.
+   */
+  column: number;
+  /**
+   * The column key name that's sorted.
+   */
+  key: string;
+  /**
+   * The direction in which the table is sorted.
+   * `undefined` represents the default unsorted state.
+   */
+  order: 'ascending' | 'descending' | undefined;
```

Comment:
```suggestion
  order?: 'ascending' | 'descending';
```

Reply:
Both would work and are acceptable right? So I would go with the more explicit style of writing out `undefined`.

Reply:
Sure, I noticed a very similar type in [`TableSorting`](https://github.com/ZEISS/precise-ui/pull/187/files#diff-f3246e6f13ea56211990d0d9f195fcc9R275) and probably it makes sense to have it consistent. It's up to you.